File: Antlr/src/main/scala/io/github/chaosunity/casc/parsing/class/Constructor.scala
package io.github.chaosunity.casc.parsing.`class`

import io.github.chaosunity.casc.parsing.`type`.{BuiltInType, Type}
import io.github.chaosunity.casc.parsing.scope.FunctionSignature
import io.github.chaosunity.casc.parsing.statement.Statement

class Constructor(signature: FunctionSignature, block: Statement) extends Function(signature, block) {
    override def returnType: Type = BuiltInType.VOID
}

File: src/main/kotlin/io/github/chaosunity/casc/parsing/Constructor.kt
package io.github.chaosunity.casc.parsing

import io.github.chaosunity.casc.parsing.node.expression.EmptyExpression
import io.github.chaosunity.casc.parsing.node.statement.Block
import io.github.chaosunity.casc.parsing.scope.AccessModifier
import io.github.chaosunity.casc.parsing.scope.FunctionSignature
import io.github.chaosunity.casc.parsing.type.BuiltInType
import io.github.chaosunity.casc.parsing.type.Type

class Constructor(signature: FunctionSignature, block: Block?, accessModifier: AccessModifier) :
    Function<Constructor>(signature, block ?: EmptyExpression(BuiltInType.VOID), accessModifier) {
    override val returnType: Type = BuiltInType.VOID
}
